cafe.rb

```diff
@@ -1,34 +1,36 @@
 # frozen_string_literal: true
 
 DRINKS = [
-  { name: 'コーヒー', price: '300' },
-  { name: 'カフェラテ', price: '400' },
-  { name: 'チャイ', price: '460' },
-  { name: 'エスプレッソ', price: '340' },
-  { name: '緑茶', price: '450' }
+  { name: 'コーヒー', price: 300 },
+  { name: 'カフェラテ', price: 400 },
+  { name: 'チャイ', price: 460 },
+  { name: 'エスプレッソ', price: 340 },
+  { name: '緑茶', price: 450 }
 ].freeze
 
 FOODS = [
-  { name: 'チーズケーキ', price: '470' },
-  { name: 'アップルパイ', price: '520' },
-  { name: 'ホットサンド', price: '410' }
+  { name: 'チーズケーキ', price: 470 },
+  { name: 'アップルパイ', price: 520 },
+  { name: 'ホットサンド', price: 410 }
 ].freeze
 
 def take_order(menus)
   menus.each.with_index(1) do |menu, i|
     puts "(#{i})#{menu[:name]}: #{menu[:price]}円"
   end
-  print '>'
-  order_number = gets.to_i
+  until (order_number ||= -1).between?(0, menus.size - 1)
+    print '>'
+    order_number = gets.to_i - 1
+  end
```

Comment:
```suggestion
  begin
    print '>'
    order_number = gets.to_i - 1
  end while menus[order_number].nil?
```

こんな感じにした方がシンプルかつコードの意図が明確になるかな？と思いますがいかがでしょう？

Reply:
レビューしていただき、ありがとうございます。
`begin...end while`のほうが読みやすいと感じます。

修正して実行してみたところ、Rubyの配列の仕様により、`order_number`が`- menus.size + 1`以上0以下の整数でも動作しました。これは許容しても良いのでしょうか?
自分としては、提示されたメニューの番号を入力すべきだと考えていますが、`begin...end until (または while !) order_number.between?(0, menus.size - 1)`では意図が伝わりにくいでしょうか?

Reply:
なるほど、たしかに`while menus[order_number].nil?`だと元のコードと挙動が異なりますね。
`end until (または while !) order_number.between?(0, menus.size - 1)` みたいな感じで良いと思います！

ちなみにGitHubのコメント通知はほかの通知メールに埋もれて気づけないときがあるので、今後は念のためFBC側のコメント欄でも「PRで質問してます」みたいな連絡をもらえると助かります〜 🙏

Reply:
> なるほど、たしかに`while menus[order_number].nil?`だと元のコードと挙動が異なりますね。 `end until (または while !) order_number.between?(0, menus.size - 1)` みたいな感じで良いと思います！

ありがとうございます!
コードを修正した後、pushします。

> ちなみにGitHubのコメント通知はほかの通知メールに埋もれて気づけないときがあるので、今後は念のためFBC側のコメント欄でも「PRで質問してます」みたいな連絡をもらえると助かります〜 🙏

承知しました。コメント欄のほうにも記入するようにします。